stdlib/asm/crypto/dsa/rpo_falcon512.masm

```diff
@@ -673,73 +583,68 @@ end
 #! via the advice stack.
 #! The signature is valid if and only if the procedure returns.
 #!
-#! Input: [PK, MSG, ...]
-#! Output: [...]
+#! Inputs:
+#!   Operand stack: [PK, MSG, ...]
+#!   Advice stack:  [tau0, tau1, h_0, ..., h_511, s2_0, ..., s2_511, pi_0, ..., pi_1022, nonce0, ..., nonce7]
+#! Outputs:
+#!   Operand stack: []
+#!   Advice stack:  []
+#!
+#! where:
+#! - h_i are the coefficients of the expanded public key polynomial.
+#! - s2_i are the coefficients of the signature polynomial.
+#! - pi_i are the coefficients of `h * s2` in Z_Q[x] where Q is the Miden VM prime.
+#! - nonce_i are field elements representing the nonce associated to the signature.
 #!
-#! Cycles: ~ 70465
-export.verify.6660
+#! Cycles: ~ 59859
+export.verify.2560
 
     # 1) Generate a Falcon signature using the secret key associated to PK on message MSG.
     adv.push_sig.rpo_falcon512
     #=> [PK, MSG, ...]
 
-    # 2) Load the NONCE from the advice provider. This is encoded as 8 field elements
-    padw adv_loadw padw adv_loadw
-    #=> [NONCE1, NONCE0, PK, MSG...]
-
-    # 3) Load the public key polynomial h and the signature polynomial s2 and the product of
+    # 2) Load the public key polynomial h and the signature polynomial s2 and the product of
     # the two polynomials pi := h * s2 in Z_Q[x]. This also checks that h hashes to the provided
-    # digest PK. While loading the polynomials, the hash of the three polynomials is computed
-    # and the first half of the digest is kept on the stack for later use by the
-    # `probabilistic_product` procedure.
+    # digest PK. While loading the polynomials, the first half of the digest of hashing the three
+    # polynomials and is used to check that we indeed have pi := h * s2 in Z_Q[x] by checking that
+    # pi(tau) = h(tau) * s2(tau) where tau is a random (Fiat-Shamir) challenge resulting
+    # from hashing h, s2 and pi.
 
-    swapdw
-    #=> [PK, MSG, NONCE1, NONCE0...]
     locaddr.0
-    #=> [h_ptr, PK, MSG, NONCE1, NONCE0...]
     exec.load_h_s2_and_product
-    #=> [tau1, tau0, tau_ptr, MSG, NONCE1, NONCE1, ...]      (Cycles: 5050)
+    #=> [MSG, ...]                                  (Cycles: 6780)
```

Comment:
Question: why does this say "6780" cycles while cycle count for `load_h_s2_and_product` procedure says "4430"?